src/Microsoft.Health.Fhir.Shared.Core/Features/Definition/SearchParameterDefinitionBuilder.cs

```diff
@@ -29,31 +29,26 @@ internal class SearchParameterDefinitionBuilder
         private readonly IModelInfoProvider _modelInfoProvider;
         private readonly Assembly _assembly;
         private readonly string _embeddedResourceName;
-        private readonly string _unsupportedParamsEmbeddedResourceName;
 
         private readonly Dictionary<Uri, SearchParameterInfo> _uriDictionary = new Dictionary<Uri, SearchParameterInfo>();
         private readonly Dictionary<string, IDictionary<string, SearchParameterInfo>> _resourceTypeDictionary = new Dictionary<string, IDictionary<string, SearchParameterInfo>>();
-        private UnsupportedSearchParameters _unsupportedParams = new UnsupportedSearchParameters();
 
         private bool _initialized;
 
         internal SearchParameterDefinitionBuilder(
             FhirJsonParser fhirJsonParser,
             IModelInfoProvider modelInfoProvider,
             Assembly assembly,
-            string embeddedResourceName,
-            string unsupportedParamsEmbeddedResourceName = null)
+            string embeddedResourceName)
         {
             EnsureArg.IsNotNull(fhirJsonParser, nameof(fhirJsonParser));
             EnsureArg.IsNotNull(modelInfoProvider, nameof(modelInfoProvider));
             EnsureArg.IsNotNull(assembly, nameof(assembly));
-            EnsureArg.IsNotNullOrWhiteSpace(embeddedResourceName, nameof(embeddedResourceName));
 
             _fhirJsonParser = fhirJsonParser;
             _modelInfoProvider = modelInfoProvider;
             _assembly = assembly;
             _embeddedResourceName = embeddedResourceName;
```

Comment:
Do we need to validate embeddedResourceName?
EnsureArg.IsNotNull(embeddedResourceName, nameof(embeddedResourceName )